packages/xlayers/src/app/editor/code-editor/editor-container/editor-container.component.ts

```diff
@@ -192,4 +193,23 @@ export class EditorContainerComponent implements OnInit, AfterContentInit {
   updateState() {
     this.store.dispatch(new CodeGen(this.codeSetting.kind, this.codeSetting.content, this.codeSetting.buttons));
   }
+
+  simulateContentEditorScroll($event) {
+    const keyCode = $event.keyCode || $event.charCode || $event.which;
+    const scroll = Math.round(0.9 * this.codeEditor.nativeElement.clientHeight);
+
+    // Page Down
+    if (keyCode === 34 || $event.key === 'PageDown' || $event.code === 'PageDown') {
```

Comment:
Instead of using the `numbers` we could use the `@angular/cdk/keycodes`.
This way we can just reference the variables (made small example below):

```
import { PAGE_DOWN  } from '@angular/cdk/keycodes';

 if (keyCode === PAGE_DOWN   || $event.key === 'PageDown' || $event.code === 'PageDown')

```


What do you think about this @sumitparakh?

Reply:
if we wanna do it the full angular way, we can use HostLiesterners

Reply:
> Instead of using the `numbers` we could use the `@angular/cdk/keycodes`.
> This way we can just reference the variables (made small example below):
> 
> ```
> import { PAGE_DOWN  } from '@angular/cdk/keycodes';
> 
>  if (keyCode === PAGE_DOWN   || $event.key === 'PageDown' || $event.code === 'PageDown')
> ```
> What do you think about this @sumitparakh?

@Jefiozie . Yeah. This is better option. Thanks. I've just updated my code.